File: sdk/dotnet/src/xjustiz.core-dotnet/Models/Entities/Core/ArtifactCore.cs
namespace xjustiz.core_dotnet.Models.Entities.Core;

using System.Text.Json.Serialization;
using System.Xml.Serialization;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Basisklasse für Abrufinformationen einer Datei.<br/>
/// <u><b>ArtifactCore:</b></u> Base class for file retrieval information.
/// </summary>
[JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
[JsonDerivedType(typeof(BundlePathArtifactCore), "bundle-path")]
[JsonDerivedType(typeof(HttpsArtifactCore), "https")]
[XmlInclude(typeof(BundlePathArtifactCore))]
[XmlInclude(typeof(HttpsArtifactCore))]
[XJustizCoreAvailability(XJustizCoreVersion.V0_2_0)]
public abstract class ArtifactCore
{
}

File: sdk/java/src/main/java/de/xjustiz/core/models/ArtifactCore.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlType;

/**
 * Basisklasse für Abrufinformationen einer Datei.
 * <p>
 * <u><b>ArtifactCore:</b></u> Base class for file retrieval information.
 */
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes({
    @JsonSubTypes.Type(value = BundlePathArtifactCore.class, name = "bundle-path"),
    @JsonSubTypes.Type(value = HttpsArtifactCore.class, name = "https")
})
@XmlType
@XmlAccessorType(XmlAccessType.FIELD)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class ArtifactCore {

    /**
     * Default constructor.
     */
    public ArtifactCore() {}

}